perun-dispatcher/src/main/java/cz/metacentrum/perun/dispatcher/processing/impl/EventProcessorImpl.java

```diff
@@ -82,41 +80,12 @@ public void run() {
 					if (event != null) {
 						if (log.isDebugEnabled()) {
 							log.debug("Events in Queue(" + eventQueue.size()
-									+ ").Dispatchers("
+									+ ").Engines("
 									+ dispatcherQueuePool.poolSize()
 									+ ").Processing event...");
 						}
-						boolean orphan = true;
-						for (DispatcherQueue dispatcherQueue : dispatcherQueuePool.getPool()) {
-							long timeStamp = 0;
-							if (log.isDebugEnabled()) {
-								timeStamp = System.currentTimeMillis();
-							}
-							if (smartMatcher.doesItMatch(event, dispatcherQueue)) {
-								orphan = false;
-								createTask(dispatcherQueue, event);
-								eventLogger.logEvent(event, dispatcherQueue.getClientID());
-								if (log.isDebugEnabled()) {
-									long timeStamp2 = System.currentTimeMillis();
-									log.debug("MATCH OK (took "
-											+ (timeStamp2 - timeStamp)
-											+ "ms) for "
-											+ dispatcherQueue.getClientID()
-											+ " AND " + event.toString());
-								}
-								break;
-							}
-							if (log.isDebugEnabled()) {
-								long timeStamp2 = System.currentTimeMillis();
-								log.debug("NO MATCH (took "
-										+ (timeStamp2 - timeStamp) + "ms) for "
-										+ dispatcherQueue.getClientID()
-										+ " AND " + event.toString());
-							}
-						}
-						if (orphan) {
-							eventLogger.logEvent(event, -1);
-						}
+						createTask(null, event);
```

Comment:
Hi, this will consequently call ``SchedulingPoolImpl.addToPool()`` and for newly created Tasks set engineId=-1. ``SchedulingQueue`` is then assigned to it later in ``scheduleTask()`` or ``sendToEngine()`` methods, but I don't see anywhere setting the engineId property back to the Task (both memory-pool and DB) once queue is resolved. Can you please check this too ? I believe correct engineId is necessary for some independent jobs which takes engineId based on queue. Thank you.